src/plugins/data/server/saved_objects/index_patterns.ts

```diff
@@ -23,7 +23,7 @@ import { indexPatternSavedObjectTypeMigrations } from './index_pattern_migration
 export const indexPatternSavedObjectType: SavedObjectsType = {
   name: 'index-pattern',
   hidden: false,
-  namespaceAgnostic: false,
```

Comment:
Shouldn't maybe `single` be the default for this option?

Reply:
You mean the `namespaceType` field, yes?

I'll refer to my comment here: https://github.com/elastic/kibana/pull/63217#discussion_r406604388

> Before #54605 was merged, the `namespaceAgnostic` field was required. Now that we are removing it, the `namespaceType` field should be required.

Perhaps the @elastic/kibana-platform team can weigh in here? I only defined this as a required field because that's how `namespaceAgnostic` was originally implemented in the New Platform.

Reply:
I'm leaning to having it mandatory too. `hidden` and `(ex)namespaceAgnostic/namespaceType ` are structuring options for the SO types, so forcing the developer to explicitly set their value seems reasonable.